### ISSUE TITLE
could not generate Deployment: expected "<image>:<tag>"

### ISSUE THREAD

**Issue by rkuccf** [Kong/kong-operator#425]
## Current Behavior
When I set kong image from my local registry eg.: localhost:5000/kong:3.7.1, I'm getting error that visible in operator logs:

```
{"level":"error","ts":"2024-07-24T14:03:32Z","msg":"Reconciler error","controller":"dataplane","controllerGroup":"gateway-operator.konghq.com","controllerKind":"DataPlane","DataPlane":{"name":"kong-s6xcs","namespace":"kong"},"namespace":"kong","name":"kong-s6xcs","reconcileID":"893e1a12-b7df-4de9-984a-00e8a5b17e1b","error":"could not build Deployment for DataPlane kong/kong-s6xcs: could not generate Deployment: expected \"<image>:<tag>\" format, got: localhost:5000/kong:4.21.1-SNAPSHOT","stacktrace":"sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:324\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:261\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.18.4/pkg/internal/controller/controller.go:222"}
```

## Expected Behavior
Want to have possibility to use kong image from my local registry during making development.

## Steps To Reproduce
1. Run local docker registry:
```shell
docker run -d -p 5000:5000 registry:2`
```
2. Push kong image to local registry: 
```shell
docker pull kong:3.7.1
docker tag kong:3.7.1 localhost:5000/kong:3.7.1
docker push localhost:5000/kong:3.7.1
```
3. Deploy the following GatewayConfiguration to your local k8s cluster:
```yaml
---
apiVersion: gateway-operator.konghq.com/v1beta1
kind: GatewayConfiguration
metadata:
  name: kong
spec:
  dataPlaneOptions:
    deployment:
      podTemplateSpec:
        spec:
          containers:
            - name: proxy
              image: "localhost:5000/kong:3.7.1"
              readinessProbe:
                initialDelaySeconds: 1
                periodSeconds: 1
  controlPlaneOptions:
    deployment:
      podTemplateSpec:
        spec:
          containers:
            - name: controller
              image: kong/kubernetes-ingress-controller:3.2.0
              env:
                - name: CONTROLLER_LOG_LEVEL
                  value: debug

``` 

## Operator Version
1.3

## `kubectl version`
Client Version: v1.28.2
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.28.2

**Comment by pmalek** [comment]
Hi @rkuccf 

For local development you can use `GATEWAY_OPERATOR_DEVELOPMENT_MODE=true` env or a corresponding cli flag which will disable the image version validation.

**Comment by rkuccf** [comment]
Thanks a lot @pmalek. It's working perfectly. Can I read something more about GATEWAY_OPERATOR_DEVELOPMENT_MODE variable? I'm wondering what other impact does variable have on operator behavior apart from disabling version validation.

**Comment by pmalek** [comment]
It's not documented (yet) but you can check the source at https://github.com/Kong/gateway-operator/blob/cc48e68d10559ac29f11b24b297711a0f3257e5d/modules/cli/cli.go or specifically at https://github.com/Kong/gateway-operator/blob/cc48e68d10559ac29f11b24b297711a0f3257e5d/modules/cli/cli.go#L155-L162

TL;DR is that currently (but that is most likely going to change):
- disabling validating webhook installation
- sending anonymous reports
- validating deployed images for CP and DP

**Comment by rkuccf** [comment]
Thanks a lot @pmalek once again.
